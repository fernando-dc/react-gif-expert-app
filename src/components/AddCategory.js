import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState("");
    
    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    };
    const handleSubmit = (e) =>{
        e.preventDefault();
        //setCategories will use the callback function to access the array elements since this component cant do that
        if (inputValue.trim().length > 2){
            setCategories( categories => [inputValue, ...categories]);
            setinputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input 
                className='animate__animated animate__fadeInDown animate__delay-1s'
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                placeholder='Input new value'
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
    