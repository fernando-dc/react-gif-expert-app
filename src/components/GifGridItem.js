import React from 'react';

export const GifGridItem = ({id, title, url}) => {
    console.log(id, title, url);
    return(
        <div className='card animate__animated animate__fadeIn animate__slow'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};
