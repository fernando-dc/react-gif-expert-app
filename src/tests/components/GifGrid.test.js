import React from 'react';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //This will mock the response of the custom hook

describe('Testing for GifGrid component', () => {

    
    test('Must match the snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,    
        });
        
        const wrapper = shallow(<GifGrid category='Gintama'/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Must show items when images are loaded using the useFetchGifts custom hook', () => {
        const gifs = [{
            id: 'FAKE_ID',
            url: 'https://localhost/fake_url.jpg',
            title: 'Fake Title',
            },
        ];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,    
        });
        const wrapper = shallow(<GifGrid category='Gintama'/>);
        
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
