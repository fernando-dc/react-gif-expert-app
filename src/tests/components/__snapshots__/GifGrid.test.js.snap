// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for GifGrid component Must match the snapshot 1`] = `
<Fragment>
  <h3>
    Gintama
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
